### PR TITLE
Manage Groups

### PR DESCRIPTION
**Pull request by mohameddahrouj**
On the server side, we will need to implement the **edit** and **delete** actions for groups.

### CODE REVIEW

File: SECP-service/src/main/resources/assets/app/directives/secp-table/secp-table.html [hunk 1]
```diff
@@ -9,11 +9,28 @@
...
+                </button>
+                <button ng-click="change(row)">
+                    <i class="fa fa-trash red" aria-hidden="true"></i>
+                </button>
+            </td>
+        </tr>
     </tbody>
 </table>
+
+<link type="text/css" rel="stylesheet" href="directives/secp-table/secp-table.css"/>
```
Comment (lavatahir) [inline SECP-service/src/main/resources/assets/app/directives/secp-table/secp-table.html]: whats up with this?

File: bower.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "angular-ui-date": "~0.0.3",
     "angular": "~1.6.7",
     "bootstrap-css": "~3.0.0",
-    "jquery": "~2.1.0",
+    "jquery": "~3.2.1",
```
Comment (lavatahir) [inline bower.json]: You made sure that the updated version of jQuery being used doesnt conflict with anything else right?

File: SECP-service/src/main/resources/assets/app/directives/secp-table/secp-table.html (hunk 1 above)
Comment (mohameddahrouj) [reply SECP-service/src/main/resources/assets/app/directives/secp-table/secp-table.html]: This is how directives work 👍

File: bower.json (hunk 2 above)
Comment (mohameddahrouj) [reply bower.json]: Correct, Tosin tested, and I also tested. jQuery is good for backwards compatibility.

File: SECP-service/src/main/java/com/visucius/secp/resources/GroupResource.java [hunk 3]
```diff
@@ -75,4 +77,17 @@ public Response deletePermissions(@Auth GroupModifyRequest request, @PathParam("
     {
         return  groupController.deletePermissions(request,id);
     }
+
+    @GET
+    @UnitOfWork
+    public Response getAllGroups() {
+        return groupController.getAllGroups();
```
Comment (afarah613) [inline SECP-service/src/main/java/com/visucius/secp/resources/GroupResource.java]: Why would a user need to get all the groups? Instead it should get all the groups for a user.

**Pull request merged**